test/modules/ModuleLib/TestResults/EdgeHubRestartDirectMethodResult.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults
+{
+    using System;
+    using System.Net;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Newtonsoft.Json;
+
+    public class EdgeHubRestartDirectMethodResult : DirectMethodTestResult
+    {
+        public EdgeHubRestartDirectMethodResult(
+            string source,
+            DateTime createdAt,
+            string trackingId,
+            Guid batchId,
+            string sequenceNumber,
+            DateTime edgeHubRestartedTime,
+            HttpStatusCode edgeHubRestartStatusCode,
+            DateTime directMethodCompletedTime,
+            HttpStatusCode directMethodCompletedStatusCode,
+            uint restartSequenceNumber)
+            : base(
+                source,
+                createdAt,
+                trackingId,
+                batchId,
+                sequenceNumber,
+                directMethodCompletedStatusCode)
+        {
+            this.EdgeHubRestartedTime = Preconditions.CheckNotNull(edgeHubRestartedTime, nameof(edgeHubRestartedTime));
+            this.EdgeHubRestartStatusCode = Preconditions.CheckNotNull(edgeHubRestartStatusCode, nameof(edgeHubRestartStatusCode));
+            this.DirectMethodCompletedTime = Preconditions.CheckNotNull(directMethodCompletedTime, nameof(directMethodCompletedTime));
+            this.DirectMethodCompletedStatusCode = Preconditions.CheckNotNull(directMethodCompletedStatusCode, nameof(directMethodCompletedStatusCode));
+            this.RestartSequenceNumber = Preconditions.CheckNotNull(restartSequenceNumber, nameof(restartSequenceNumber));
+        }
+
+        DateTime EdgeHubRestartedTime { get; set; }
+
+        public HttpStatusCode EdgeHubRestartStatusCode { get; set; }
```

Comment:
why do we need this?

Reply:
I thought I'd have a result as `failed` but a tester can take a look at it himself what's going wrong.

Reply:
1. Remove the restart status 
2. Make sure the restart always  success before continue w/ the test.